sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryStatusAndProgressSuite.scala

```diff
@@ -38,13 +38,18 @@ class StreamingQueryStatusAndProgressSuite extends SparkFunSuite {
         |  "id" : "${testProgress1.id.toString}",
         |  "runId" : "${testProgress1.runId.toString}",
         |  "name" : "myName",
-        |  "timestamp" : "2016-12-05T20:54:20.827Z",
+        |  "triggerTimestamp" : "2016-12-05T20:54:20.827Z",
         |  "numInputRows" : 678,
         |  "inputRowsPerSecond" : 10.0,
         |  "durationMs" : {
         |    "total" : 0
         |  },
-        |  "currentWatermark" : 3,
+        |  "queryTimestamps" : {
+        |    "eventTime.avg" : "2016-12-05T20:54:20.827Z",
```

Comment:
1000 *  5 seconds batch is < 2 hours. Anyways, I think it will be equally or more confusing adding it later as a top-level field in Progress (e.g. `timestamp`, `eventTime`, `processingTime` ?). It may be better to have the top-level `timestamp`, and all other execution level detailed timestamps inside a single map. 

Anyways, I am updating the PR, but I think this is a little short-sighted.